components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -98,7 +98,7 @@ class DownloadsFeature(
                     if (download != null) {
                         // Update the file name to ensure it doesn't collide with one already on disk]
                         val downloadWithUniqueName = download.fileName?.let {
-                            download.copy(fileName = uniqueFileName(
+                            download.copy(fileName = DownloadUtils.uniqueFileName(
```

Comment:
@sblatz I missed this in my first review, but @pocmo found it: We shouldn't change the state of the session here. 

The `DownloadState` describes a request to download a file with the name provided by the engine, which is different from the unique file name dependent on the file system on the device i.e. two sessions could download the same file and we can only determine the local file name once we write to disk. So, we should determine the unique file name in the download service when we perform the download.

Reply:
Ah, scratch the updating download state part, as we're only copying here to forward an updated object to the download manager / service. I think it would still be cleaner to do this in the service instead (when we open / start writing to the file)?

Reply:
@csadilek I opted for creating a unique name for the file in `DownloadFeature` because I want the `DownloadPrompt` to display the correct name. That means even **before** a user is writing to that file, we've determined a unique name for the file. This matches the behavior of desktop.

Reply:
Actually I'm wrong here. This is something that desktop does *not* do. It just seemed to make sense to me that we would show the user the file as its actual name in the prompt.

Reply:
I have moved this logic to the service right before writing the file 😄

Reply:
Hmm. Seems moving the logic to the service causes file corruption when the files conflict. Can we keep it inside of the `DownloadsFeature` and file that as a follow-up?

Reply:
OK, looks good now!